lib/bloc/location/location_bloc.dart

```diff
@@ -15,21 +17,104 @@ class LocationBloc extends Bloc<LocationEvent, LocationState> {
         (LocationEvent event, Emitter<LocationState> emit) async {
       await _requestCurrentLocation(emit);
     });
+    on<RequestPermissionLocationEvent>(
+        (LocationEvent event, Emitter<LocationState> emit) async {
+      await _requestPermissionLocationEvent(emit);
+    });
+    on<DefaultLocationServiceEvent>(
+        (LocationEvent event, Emitter<LocationState> emit) async {
+      await _defaultLocationServiceEvent(emit);
+    });
   }
 
   Future<void> _requestCurrentLocation(Emitter<LocationState> emit) async {
-    try {
-      final LocationDetails data = await _locationService.getCurrentLocation();
-      final String dateDisplay =
-          DateFormat('dd-MMM-yyyy').format(DateTime.now());
+    final bool isConnected = await InternetConnectionChecker().hasConnection;
```

Comment:
Internet connection is not needed to get user's current location. Please remove this part